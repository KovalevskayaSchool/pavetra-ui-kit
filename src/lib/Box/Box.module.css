.box {
  --bg-color-box: var(--pv-color-white);
  border-radius: var(--pv-border-radius-l);
  background-color: var(--bg-color-box);
  box-sizing: border-box;
}
