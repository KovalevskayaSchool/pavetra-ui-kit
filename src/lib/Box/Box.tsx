import {
  forwardRef,
  HTMLAttributes,
  PropsWithChildren,
  Ref,
  ElementType,
} from "react";
import cn from "classnames";

import "./Box.css";

export interface BoxProps extends Partial<HTMLAttributes<HTMLDivElement>> {
  ref?: Ref<HTMLElement>;
  as?: ElementType;
}

export const Box = forwardRef<HTMLElement, PropsWithChildren<BoxProps>>(
  ({ className, children, as = "div", ...props }, ref) => {
    const Component = as;
    return (
      <Component ref={ref} {...props} className={cn("ks-box", className)}>
        {children}
      </Component>
    );
  }
);

Box.displayName = "Box";
