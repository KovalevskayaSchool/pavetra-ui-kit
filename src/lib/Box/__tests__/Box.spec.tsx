import React from "react";
import { render } from "@testing-library/react";

import { Box } from "../Box";

describe("Box", () => {
  test("should render component", () => {
    const { container } = render(<Box />);

    expect(container).toMatchSnapshot();
  });
});
