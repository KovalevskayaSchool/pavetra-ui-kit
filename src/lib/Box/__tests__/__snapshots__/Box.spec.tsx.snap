// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render component 1`] = `
<div>
  <div
    class="box"
  />
</div>
`;
